### PR TITLE
FQDNS- New test and fixes

### PR DESCRIPTION
**Pull request by eloycoto**
Hi, 

Two commits here: 

- Roundrobin: The test didn't apply the policy correctly and no enforce at all. 
- New CNAME follow test, to validate that the CNAME entries are working correctly.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6327)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/fqdn.go [hunk 1]
```diff
@@ -421,6 +440,99 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
...
+	}
+]`
+
+		_, err = vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+		target = "http://level3CNAME.cilium.test"
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(target))
+		res.ExpectSuccess("Container %q cannot access to %q when should work", helpers.App1, target)
```
Comment (raybejjani) [inline test/runtime/fqdn.go]: `helpers.App2`, no? for both the comment and the exec? unless you wanted the policy to apply to app1

File: test/runtime/fqdn.go [hunk 2]
```diff
@@ -411,6 +420,16 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
...
 		Expect(err).To(BeNil(), "Policy cannot be imported")
 
+		endpoints, err := vm.GetEndpointsIds()
+		Expect(err).To(BeNil(), "Endpoints can't be retrieved")
+
+		for _, container := range []string{helpers.App1, helpers.App2} {
+			ep := vm.EndpointGet(endpoints[container])
+			Expect(ep.Status.Policy.Realized.PolicyEnabled).To(
+				Equal(models.EndpointPolicyEnabledEgress),
+				"Endpoint %q do not have policy applied", container)
```
Comment (raybejjani) [inline test/runtime/fqdn.go]: "does not have a policy applied"

**Review by raybejjani — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

File: test/runtime/fqdn.go [hunk 3]
```diff
@@ -420,6 +420,16 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
 		_, err := vm.PolicyRenderAndImport(policy)
 		Expect(err).To(BeNil(), "Policy cannot be imported")
 
+		endpoints, err := vm.GetEndpointsIds()
```
Comment (ianvernon) [inline test/runtime/fqdn.go:423]: `endpoints` is never used again after this. Why not do the assertion for each value of `ep` in the for loop below?

File: test/runtime/fqdn.go [hunk 4]
```diff
@@ -420,6 +420,16 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
...
+		Expect(err).To(BeNil(), "Endpoints can't be retrieved")
+
+		for _, container := range []string{helpers.App1, helpers.App2} {
+			ep := vm.EndpointGet(endpoints[container])
+			Expect(ep.Status.Policy.Realized.PolicyEnabled).To(
```
Comment (ianvernon) [inline test/runtime/fqdn.go:432]: This will panic if the endpoints are nil.

**Review by ianvernon — CHANGES_REQUESTED** [review]
One small fix needed.

File: test/runtime/fqdn.go (hunk 3 above)
Comment (eloycoto) [reply test/runtime/fqdn.go:423]: I'm not sure If I understand this one, endpoints is a map with containerName:epID and it's used in the loop.

File: test/runtime/fqdn.go (hunk 4 above)
Comment (eloycoto) [reply test/runtime/fqdn.go:432]: Nice one, checking the key just before of this one :-) Thanks

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Pull request merged**